lib/ansible/modules/cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,1765 @@
+                                  **camel_dict_to_snake_dict(e.response))
+
+    def remove_all_tags_from_resource(self, arn):
+        tags = self.list_tags_for_resource(arn)
+        key_list = []
+        for tag in tags:
+            key_list.append(tag.get('Key'))
+        self.untag_resource(arn, key_list)
+
+    def list_tags_for_resource(self, arn):
+        try:
+            func = partial(self.client.list_tags_for_resource, Resource=arn)
+            response = self.paginated_response(func)
+            return response.get('Tags').get('Items')
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(msg="error removing all tags from resource - " + str(e), exception=traceback.format_exc(),
+                                  **camel_dict_to_snake_dict(e.response))
+
+    def paginated_response(self, func, result_key=''):
+        '''
+        Returns expanded response for paginated operations.
+        The 'result_key' is used to define the concatenated results that are combined
+        from each paginated response.
+        '''
+        args = dict()
+        results = dict()
+        loop = True
+        while loop:
+            response = func(**args)
+            if result_key == '':
+                result = response
+                result.pop('ResponseMetadata', None)
+            else:
+                result = response.get(result_key)
+            results.update(result)
+            args['NextToken'] = response.get('NextToken')
+            loop = args['NextToken'] is not None
+        return results
+
+
+class CloudFrontValidationManager:
+    """
+    Manages Cloudfront validations
+    """
+
+    def __init__(self, module):
+        self.__cloudfront_facts_mgr = CloudFrontFactsServiceManager(module)
+        self.module = module
+        self.__helpers = CloudFrontHelpers()
+        self.__default_distribution_enabled = True
+        self.__default_http_port = 80
+        self.__default_https_port = 443
+        self.__default_is_ipv6_enabled = False
+        self.__default_origin_ssl_protocols = ['TLSv1', 'TLSv1.1', 'TLSv1.2']
+        self.__default_custom_origin_protocol_policy = 'match-viewer'
+        self.__default_datetime_string = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')
+        self.__default_cache_behavior_min_ttl = 0
+        self.__default_cache_behavior_max_ttl = 31536000
+        self.__default_cache_behavior_default_ttl = 86400
+        self.__default_cache_behavior_trusted_signers_enabled = False
+        self.__default_cache_behavior_compress = False
+        self.__default_cache_behavior_viewer_protocol_policy = 'allow-all'
+        self.__default_cache_behavior_smooth_streaming = False
+        self.__default_cache_behavior_forwarded_values_forward_cookies = 'none'
+        self.__default_cache_behavior_forwarded_values_query_string = True
+        self.__default_trusted_signers_enabled = True
```

Comment:
Shouldn't this default to `False` like in line 896?  This is causing distributions to come up with trusted signers enabled but no trusted signers configured by default.